tests/test_literals.py

```diff
@@ -1,34 +1,56 @@
 """Test RDF literals."""
-# pylint: disable=no-member
-from triplestore import en, Literal, XSD
-
-
-l1 = Literal("Hello world!")
-assert l1 == "Hello world!"
-assert isinstance(l1, str)
-assert l1.lang is None
-assert l1.datatype is None
-assert l1.to_python() == "Hello world!"
-assert l1.value == "Hello world!"
-assert l1.n3() == '"Hello world!"'
-
-l2 = Literal("Hello world!", lang="en")
-assert l2.lang == "en"
-assert l2.datatype is None
-assert l2.value == "Hello world!"
-assert l2.n3() == '"Hello world!"@en'
-
-l3 = en("Hello world!")
-assert l3.n3() == '"Hello world!"@en'
-
-l4 = Literal(42)
-assert l4.lang is None
-assert l4.datatype == XSD.integer
-assert l4.value == 42
-assert l4.n3() == f'"42"^^{XSD.integer}'
-
-l5 = Literal(42, datatype=float)
-assert l5.lang is None
-assert l5.datatype == XSD.double
-assert l5.value == 42.0
-assert l5.n3() == f'"42"^^{XSD.double}'
+
```

Comment:
I have difficulties to see the benefits of putting all tests into functions.

The old script is easier to read and so much easier to play around with during development or debugging in an good interactive interpreter like ipython.

Reply:
It's to achieve proper test isolation to not have unexpected setups occurring, resulting in unintended and opaque environments when testing certain aspects, unknowingly performing faulty tests, since you're not specifically testing a functionality in a _fully known_ environment. Any failing tests will be difficult to debug and understand, more so the later in the script they appear.

The tests should be minimalistic enough for you to _not_ have to create breakpoints, but rather understand what part of the code is failing straight away. This is the point of unit testing as I understand it.
However, a functionality in the code base might depend on another part of the code base (most likely), in which case a proper debugging environment is necessary and should be something we should strive towards and force ourselves to implement.
Writing code without a proper debugging setup is like driving a car without a steering wheel.

It is true the pytest framework has a specific way of working and running. And I'm happy to consider another modern framework if another is suggested. However, I think it may be faster for you to learn this system than it is for everyone to learn a new one? But it might indeed be time well invested and spent.

If you crave the usage of script-based testing locally, I suggest to create _proper_ example scripts that can be used in the documentation simultaneously as for your local testing.
These can for example be placed under `/docs/examples/`.

Reply:
I strongly agree with Casper. Pytest is a pretty standard framework these days. (And it is not that fundamentally different from the standard library unittest https://docs.python.org/3/library/unittest.html).  I have used breakpoints all the time and it is not *that* difficult. Agree as well that the purpose of tests is to test the code, it is not a set of user-friendly tutorials, which should be made in a separate example session.

Reply:
I would normally argue to use a standard framework. It is not that I don't understand pytest or are able to use it (like VS Code), but I have difficulties to see the real value it provides - except for test isolation, that is actually a good argument. There are also some powerful test fixtures, which might be nice when you are writing the tests, but insane if you try to understand a test someone else has written. Documentation of which conftest.py file or builtin pytest module that implements the fixture may mitigate this. But so far I haven't seen any such documentation.

On the other hand, I see when I am co-developing with others, how powerful it is for quick checks during development, to write your test script in parallel with the code and run it in a good interactive interpreter in a terminal next to your text editor. You can't do that with pytest test functions. 

The suggestion of Casper with an example script in docs/examples/ may be a compromise. That may even be a good way to force you to think about and good use case and user friendly interface for the library while writing the code.